clients/src/main/java/org/apache/kafka/common/requests/LeaveGroupResponse.java

```diff
@@ -17,47 +17,126 @@
 package org.apache.kafka.common.requests;
 
 import org.apache.kafka.common.message.LeaveGroupResponseData;
+import org.apache.kafka.common.message.LeaveGroupResponseData.MemberResponse;
 import org.apache.kafka.common.protocol.ApiKeys;
 import org.apache.kafka.common.protocol.Errors;
 import org.apache.kafka.common.protocol.types.Struct;
 
 import java.nio.ByteBuffer;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
+/**
+ * Possible error codes.
+ *
+ * Top level errors:
+ * - {@link Errors#COORDINATOR_LOAD_IN_PROGRESS}
+ * - {@link Errors#COORDINATOR_NOT_AVAILABLE}
+ * - {@link Errors#NOT_COORDINATOR}
+ * - {@link Errors#GROUP_AUTHORIZATION_FAILED}
+ * - {@link Errors#UNKNOWN_MEMBER_ID} returned when the group is not found
+ *
+ * Member level errors:
+ * - {@link Errors#FENCED_INSTANCE_ID}
+ * - {@link Errors#UNKNOWN_MEMBER_ID}
+ */
 public class LeaveGroupResponse extends AbstractResponse {
 
-    private final LeaveGroupResponseData data;
+    public final LeaveGroupResponseData data;
 
     public LeaveGroupResponse(LeaveGroupResponseData data) {
         this.data = data;
     }
 
+    public LeaveGroupResponse(List<MemberResponse> memberResponses,
+                              Errors topLevelError,
+                              final int throttleTimeMs,
+                              final short version) {
+        if (version <= 2) {
+            if (memberResponses.size() != 1) {
```

Comment:
It seems a little inconsistent to require more than one member only when the version is old. But for old versions of the API, the top-level error code in fact is all we need. So I'm thinking we can remove this check?

Reply:
We still need to handle `UNKNOWN_MEMBER_ID` case for old single member API I guess. So member level error is still possible and this strict check helps preventing bugs in getting wrong number of member response.

Reply:
My point is that it's arbitrary. As long as we have a top level error code, we can still fulfill the old leave group response. So why not just make the behavior consistent whether it is the old version or the new version?